docs/architecture/adr/0025-ts-deprecate-enums.md

```diff
@@ -0,0 +1,196 @@
+
+```ts
+// declare the raw data and reduce repetition with an internal type
+const _CipherType = {
+  Login: 1,
+  SecureNote: 2,
+  Card: 3,
+  Identity: 4,
+  SshKey: 5,
+} as const;
+
+type _CipherType = typeof _CipherType;
+
+// derive the enum-like type from the raw data
+export type CipherType = _CipherType[keyof _CipherType];
+
+// assert that the raw data is of the enum-like type
+export const CipherType: Readonly<{ [K in keyof _CipherType]: CipherType }> =
+  Object.freeze(_CipherType);
+```
+
+This code creates a `type CipherType` that allows arguments and variables to be typed similarly to
+an enum. It also strongly types the `const CiperType` so that direct accesses of its members
+preserve type safety. This ensures that type inference properly limits the accepted values to those
+allowed by `type CipherType`. Without the type assertion, the compiler infers `number` in these
+cases:
+
+```ts
+const s = new Subject(CipherType.Login); // `s` is a `Subject<CipherType>`
+const a = [CipherType.Login, CipherType.Card]; // `a` is an `Array<CipherType>`
+const m = new Map([[CipherType.Login, ""]]); // `m` is a `Map<CipherType, string>`
+```
+
+:::warning
+
+- Types that use enums like [computed property names][computed-property-names] issue a compiler
+  error with this pattern. [This issue is fixed as of TypeScript 5.8][no-member-fields-fixed].
+- Certain objects are more difficult to create with this pattern. This is explored in
+  [Appendix A](#appendix-a-mapped-types-and-enum-likes).
+
+:::
+
+## Considered Options
+
+- **Allow enums, but advise against them** - This is the current state of affairs. With this option,
+  teams **must** address the FIXMEs, but _may_ address them by disabling the lint.
+- **Deprecate enum use** - Allow enums to exist for historic or technical purposes, but prohibit the
+  introduction of new ones. Increase the lint to a "warning" and allow the lint to be disabled.
+- **Eliminate enum use** - Prohibit the introduction of any new enum and replace all enums in the
+  codebase with typescript objects. Increase the lint to an "error" and prohibit disabling of the
+  lint.
+
+## Decision Outcome
+
+Chosen option: **Deprecate enum use**
+
+### Positive Consequences
+
+- Allows for cases where autogenerated code introduces an enum by necessity.
+- Developers receive a warning in their IDE to discourage new enums.
+- The warning can direct them to our contributing docs, where they can learn typesafe alternatives.
+- Our compiled code size decreases when enums are replaced.
+- If all teams eliminate enums in practice, the warning can be increased to an error.
+
+### Negative Consequences
+
+- Unnecessary usage may persist indefinitely on teams carrying a high tech debt.
+- The lint increased the number of FIXME comments in the code by about 10%.
+
+### Plan
+
+- Update contributing docs with patterns and best practices for enum replacement.
+- Update the reporting level of the lint to "warning".
```

Comment:
Or are you maybe suggesting a change to our current approach, i.e. to downgrade error -> warning